<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Discover motifs in sequences — denovo_motifs • MotifPeeker</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Discover motifs in sequences — denovo_motifs"><meta name="description" content="Use STREME from MEME suite to find  motifs in the provided sequences.
To speed up the process, the sequences can be optionally trimmed to reduce
the search space. The result is then optionally filtered to remove motifs
with a high number of nucleotide repeats"><meta property="og:description" content="Use STREME from MEME suite to find  motifs in the provided sequences.
To speed up the process, the sequences can be optionally trimmed to reduce
the search space. The result is then optionally filtered to remove motifs
with a high number of nucleotide repeats"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotifPeeker</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.11</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MotifPeeker.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/docker.html">Docker/Singularity Containers</a></li>
    <li><a class="dropdown-item" href="../articles/examples.html">Example Reports</a></li>
    <li><a class="dropdown-item" href="../articles/troubleshooting.html">Troubleshooting</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/neurogenomics/MotifPeeker/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Discover motifs in sequences</h1>
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MotifPeeker/blob/master/R/denovo_motifs.R" class="external-link"><code>R/denovo_motifs.R</code></a></small>
      <div class="d-none name"><code>denovo_motifs.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Use STREME from MEME suite to find  motifs in the provided sequences.
To speed up the process, the sequences can be optionally trimmed to reduce
the search space. The result is then optionally filtered to remove motifs
with a high number of nucleotide repeats</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">denovo_motifs</span><span class="op">(</span></span>
<span>  <span class="va">seqs</span>,</span>
<span>  <span class="va">trim_seq_width</span>,</span>
<span>  <span class="va">genome_build</span>,</span>
<span>  discover_motifs_count <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  minw <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  maxw <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  filter_n <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  meme_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  debug <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seqs">seqs<a class="anchor" aria-label="anchor" href="#arg-seqs"></a></dt>
<dd><p>A list of <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> objects containing
sequences to search for motifs.</p></dd>


<dt id="arg-trim-seq-width">trim_seq_width<a class="anchor" aria-label="anchor" href="#arg-trim-seq-width"></a></dt>
<dd><p>An integer specifying the width of the sequence to
extract around the summit (default = NULL). This sequence is used to search
for discovered motifs. If not provided, the entire peak region will be used.
This parameter is intended to reduce the search space and speed up motif
discovery; therefore, a value less than the average peak width is
recommended. Peaks are trimmed symmetrically around the summit while
respecting the peak bounds.</p></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>The genome build that the peak sequences should be
derived from.</p></dd>


<dt id="arg-discover-motifs-count">discover_motifs_count<a class="anchor" aria-label="anchor" href="#arg-discover-motifs-count"></a></dt>
<dd><p>An integer specifying the number of motifs to
discover. (default = 3) Note that higher values take longer to compute.</p></dd>


<dt id="arg-minw">minw<a class="anchor" aria-label="anchor" href="#arg-minw"></a></dt>
<dd><p>An integer specifying the minimum width of the motif.
(default = 8)</p></dd>


<dt id="arg-maxw">maxw<a class="anchor" aria-label="anchor" href="#arg-maxw"></a></dt>
<dd><p>An integer specifying the maximum width of the motif.
(default = 25)</p></dd>


<dt id="arg-filter-n">filter_n<a class="anchor" aria-label="anchor" href="#arg-filter-n"></a></dt>
<dd><p>An integer specifying the number of consecutive nucleotide
repeats a discovered motif must contain to be filtered out.
(default = 6)</p></dd>


<dt id="arg-out-dir">out_dir<a class="anchor" aria-label="anchor" href="#arg-out-dir"></a></dt>
<dd><p>A <code>character</code> vector of output directory to save STREME
results to. (default = <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>)</p></dd>


<dt id="arg-meme-path">meme_path<a class="anchor" aria-label="anchor" href="#arg-meme-path"></a></dt>
<dd><p>path to "meme/bin/" (default: <code>NULL</code>). Will use default
search behavior as described in <code>check_meme_install()</code> if unset.</p></dd>


<dt id="arg-bpparam">BPPARAM<a class="anchor" aria-label="anchor" href="#arg-bpparam"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html" class="external-link">BiocParallelParam-class</a></code> object
specifying run parameters. (default = SerialParam(), single core run)</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A logical indicating whether to print verbose messages while
running the function. (default = FALSE)</p></dd>


<dt id="arg-debug">debug<a class="anchor" aria-label="anchor" href="#arg-debug"></a></dt>
<dd><p>A logical indicating whether to print debug messages while
running the function. (default = FALSE)</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments to pass to <code>STREME</code>. For more
information, refer to the official MEME Suite documentation on
<a href="https://meme-suite.org/meme/doc/streme.html" class="external-link">STREME</a>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list of <code><a href="https://rdrr.io/pkg/universalmotif/man/universalmotif-class.html" class="external-link">universalmotif</a></code> objects and
associated metadata.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu">memes</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/memes/man/meme_is_installed.html" class="external-link">meme_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CTCF_TIP_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">genome_build</span> <span class="op">&lt;-</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span></span></span>
<span class="r-in"><span>   </span></span>
<span class="r-in"><span>    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">denovo_motifs</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CTCF_TIP_peaks</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                        trim_seq_width <span class="op">=</span> <span class="fl">50</span>,</span></span>
<span class="r-in"><span>                        genome_build <span class="op">=</span> <span class="va">genome_build</span>,</span></span>
<span class="r-in"><span>                        discover_motifs_count <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                        filter_n <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>                        minw <span class="op">=</span> <span class="fl">8</span>,</span></span>
<span class="r-in"><span>                        maxw <span class="op">=</span> <span class="fl">8</span>,</span></span>
<span class="r-in"><span>                        out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">consensus</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: p-values will be inaccurate if primary and control</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: p-values will be inaccurate if primary and control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "TCCCAKMACN"</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hiranyamaya Dash, Thomas Roberts, Nathan Skene.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

