<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get dataframe with motif enrichment values — get_df_enrichment • MotifPeeker</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get dataframe with motif enrichment values — get_df_enrichment"><meta name="description" content="Wrapper for `MotifPeeker::motif_enrichment` to get motif enrichment counts
and percentages for all peaks and motifs, generating a data.frame
suitable for plots. The data.frame contains values for all and
segregated peaks."><meta property="og:description" content="Wrapper for `MotifPeeker::motif_enrichment` to get motif enrichment counts
and percentages for all peaks and motifs, generating a data.frame
suitable for plots. The data.frame contains values for all and
segregated peaks."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotifPeeker</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MotifPeeker.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/docker.html">Docker/Singularity Containers</a></li>
    <li><a class="dropdown-item" href="../articles/troubleshooting.html">Troubleshooting</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/neurogenomics/MotifPeeker/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get dataframe with motif enrichment values</h1>
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MotifPeeker/blob/master/R/get_df_enrichment.R" class="external-link"><code>R/get_df_enrichment.R</code></a></small>
      <div class="d-none name"><code>get_df_enrichment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Wrapper for `MotifPeeker::motif_enrichment` to get motif enrichment counts
and percentages for all peaks and motifs, generating a <code>data.frame</code>
suitable for plots. The <code>data.frame</code> contains values for all and
segregated peaks.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_df_enrichment</span><span class="op">(</span></span>
<span>  <span class="va">result</span>,</span>
<span>  <span class="va">segregated_peaks</span>,</span>
<span>  <span class="va">user_motifs</span>,</span>
<span>  <span class="va">genome_build</span>,</span>
<span>  reference_index <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  workers <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  meme_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-result">result<a class="anchor" aria-label="anchor" href="#arg-result"></a></dt>
<dd><p>A <code>list</code> with the following elements:</p><dl><dt>peaks</dt>
<dd><p>A <code>list</code> of peak files generated using
<code><a href="read_peak_file.html">read_peak_file</a></code>.</p></dd>

    <dt>alignments</dt>
<dd><p>A <code>list</code> of alignment files.</p></dd>

    <dt>exp_type</dt>
<dd><p>A <code>character</code> vector of experiment types.</p></dd>

    <dt>exp_labels</dt>
<dd><p>A <code>character</code> vector of experiment labels.</p></dd>

    <dt>read_count</dt>
<dd><p>A <code>numeric</code> vector of read counts.</p></dd>

    <dt>peak_count</dt>
<dd><p>A <code>numeric</code> vector of peak counts.</p></dd>


</dl></dd>


<dt id="arg-segregated-peaks">segregated_peaks<a class="anchor" aria-label="anchor" href="#arg-segregated-peaks"></a></dt>
<dd><p>A <code>list</code> object generated using
<code><a href="segregate_seqs.html">segregate_seqs</a></code>.</p></dd>


<dt id="arg-user-motifs">user_motifs<a class="anchor" aria-label="anchor" href="#arg-user-motifs"></a></dt>
<dd><p>A <code>list</code> with the following elements:</p><dl><dt>motifs</dt>
<dd><p>A <code>list</code> of motif files.</p></dd>

    <dt>motif_labels</dt>
<dd><p>A <code>character</code> vector of motif labels.</p></dd>


</dl></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>A character string with the abbreviated genome build
name, or a BSGenome object. At the moment, only <code>hg38</code> and <code>hg19</code>
are supported as abbreviated input.</p></dd>


<dt id="arg-reference-index">reference_index<a class="anchor" aria-label="anchor" href="#arg-reference-index"></a></dt>
<dd><p>An integer specifying the index of the peak file to
use as the reference dataset for comparison. Indexing starts from 1.
(default = 1)</p></dd>


<dt id="arg-out-dir">out_dir<a class="anchor" aria-label="anchor" href="#arg-out-dir"></a></dt>
<dd><p>A <code>character</code> vector of output directory.</p></dd>


<dt id="arg-workers">workers<a class="anchor" aria-label="anchor" href="#arg-workers"></a></dt>
<dd><p>An integer specifying the number of threads to use for
parallel processing. (default = 1)<br><strong>IMPORTANT:</strong> For each worker, please ensure a minimum of 6GB of
memory (RAM) is available as `denovo_motif_discovery` is memory-intensive.</p></dd>


<dt id="arg-meme-path">meme_path<a class="anchor" aria-label="anchor" href="#arg-meme-path"></a></dt>
<dd><p>path to <code>meme/bin/</code> (optional). Defaut: <code>NULL</code>, searches
"MEME_PATH" environment variable or "meme_path" option for path to "meme/bin/".</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A logical indicating whether to print verbose messages while
running the function. (default = FALSE)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code>data.frame</code> with the following columns:</p><dl><dt>exp_label</dt>
<dd><p>Experiment labels.</p></dd>

 <dt>exp_type</dt>
<dd><p>Experiment types.</p></dd>

 <dt>motif_indice</dt>
<dd><p>Motif indices.</p></dd>

 <dt>group1</dt>
<dd><p>Segregated group- "all", "Common" or "Unique".</p></dd>

 <dt>group2</dt>
<dd><p>"reference" or "comparison" group.</p></dd>

 <dt>count_enriched</dt>
<dd><p>Number of peaks with motif.</p></dd>

 <dt>count_nonenriched</dt>
<dd><p>Number of peaks without motif.</p></dd>

 <dt>perc_enriched</dt>
<dd><p>Percentage of peaks with motif.</p></dd>

 <dt>perc_nonenriched</dt>
<dd><p>Percentage of peaks without motif.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other generate data.frames:
<code><a href="get_df_distances.html">get_df_distances</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CTCF_ChIP_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CTCF_TIP_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"motif_MA1102.3"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"motif_MA1930.2"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CTCF_ChIP_peaks</span>, <span class="va">CTCF_TIP_peaks</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    exp_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ChIP"</span>, <span class="st">"TIP"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    exp_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CTCF_ChIP"</span>, <span class="st">"CTCF_TIP"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    read_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">150</span>, <span class="fl">200</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    peak_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">120</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">segregated_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="segregate_seqs.html">segregate_seqs</a></span><span class="op">(</span><span class="va">input</span><span class="op">$</span><span class="va">peaks</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">input</span><span class="op">$</span><span class="va">peaks</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    motifs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">motif_MA1930.2</span>, <span class="va">motif_MA1102.3</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    motif_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"MA1930.2"</span>, <span class="st">"MA1102.3"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">reference_index</span> <span class="op">&lt;-</span> <span class="fl">1</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">genome_build</span> <span class="op">&lt;-</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span></span></span>
<span class="r-in"><span>    </span></span>
<span class="r-in"><span>    <span class="va">enrichment_df</span> <span class="op">&lt;-</span> <span class="fu">get_df_enrichment</span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="va">input</span>, <span class="va">segregated_input</span>, <span class="va">motifs</span>, <span class="va">genome_build</span>, reference_index <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>        workers <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hiranyamaya Dash, Thomas Roberts, Nathan Skene.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

