<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ MotifPeeker</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotifPeeker</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MotifPeeker.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/docker.html">Docker/Singularity Containers</a></li>
    <li><a class="dropdown-item" href="../articles/examples.html">Example Reports</a></li>
    <li><a class="dropdown-item" href="../articles/troubleshooting.html">Troubleshooting</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/neurogenomics/MotifPeeker/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="CTCF_ChIP_peaks.html">CTCF_ChIP_peaks</a></code>

        </dt>
        <dd>Example ChIP-seq peak file</dd>
      </dl><dl><dt>

          <code><a href="CTCF_TIP_peaks.html">CTCF_TIP_peaks</a></code>

        </dt>
        <dd>Example TIP-seq peak file</dd>
      </dl><dl><dt>

          <code><a href="MotifPeeker.html">MotifPeeker()</a></code>

        </dt>
        <dd>Benchmark epigenomic profiling methods using motif enrichment</dd>
      </dl><dl><dt>

          <code><a href="bpapply.html">bpapply()</a></code>

        </dt>
        <dd>Use BiocParallel functions with appropriate parameters</dd>
      </dl><dl><dt>

          <code><a href="calc_frip.html">calc_frip()</a></code>

        </dt>
        <dd>Calculate FRiP score</dd>
      </dl><dl><dt>

          <code><a href="check_ENCODE.html">check_ENCODE()</a></code>

        </dt>
        <dd>Check for ENCODE input</dd>
      </dl><dl><dt>

          <code><a href="check_JASPAR.html">check_JASPAR()</a></code>

        </dt>
        <dd>Check for JASPAR input</dd>
      </dl><dl><dt>

          <code><a href="check_genome_build.html">check_genome_build()</a></code>

        </dt>
        <dd>Check genome build</dd>
      </dl><dl><dt>

          <code><a href="denovo_motifs.html">denovo_motifs()</a></code>

        </dt>
        <dd>Find de-novo motifs in sequences</dd>
      </dl><dl><dt>

          <code><a href="find_motifs.html">find_motifs()</a></code>

        </dt>
        <dd>Find similar motifs</dd>
      </dl><dl><dt>

          <code><a href="format_exptype.html">format_exptype()</a></code>

        </dt>
        <dd>Format exp_type</dd>
      </dl><dl><dt>

          <code><a href="get_JASPARCORE.html">get_JASPARCORE()</a></code>

        </dt>
        <dd>Download JASPAR CORE database</dd>
      </dl><dl><dt>

          <code><a href="get_df_distances.html">get_df_distances()</a></code>

        </dt>
        <dd>Get dataframe with motif-summit distances</dd>
      </dl><dl><dt>

          <code><a href="get_df_enrichment.html">get_df_enrichment()</a></code>

        </dt>
        <dd>Get dataframe with motif enrichment values</dd>
      </dl><dl><dt>

          <code><a href="motif_MA1102.3.html">motif_MA1102.3</a></code>

        </dt>
        <dd>Example CTCFL JASPAR motif file</dd>
      </dl><dl><dt>

          <code><a href="motif_MA1930.2.html">motif_MA1930.2</a></code>

        </dt>
        <dd>Example CTCF JASPAR motif file</dd>
      </dl><dl><dt>

          <code><a href="motif_enrichment.html">motif_enrichment()</a></code>

        </dt>
        <dd>Calculate motif enrichment in a set of sequences</dd>
      </dl><dl><dt>

          <code><a href="motif_similarity.html">motif_similarity()</a></code>

        </dt>
        <dd>Compare motifs from segregated sequences</dd>
      </dl><dl><dt>

          <code><a href="plot_enrichment_individual.html">plot_enrichment_individual()</a></code>

        </dt>
        <dd>Plot motif-enrichment for individual experiments</dd>
      </dl><dl><dt>

          <code><a href="plot_enrichment_overall.html">plot_enrichment_overall()</a></code>

        </dt>
        <dd>Plot motif-enrichment for all experiments</dd>
      </dl><dl><dt>

          <code><a href="pretty_number.html">pretty_number()</a></code>

        </dt>
        <dd>Convert numbers to more readable strings</dd>
      </dl><dl><dt>

          <code><a href="read_motif_file.html">read_motif_file()</a></code>

        </dt>
        <dd>Read a motif file</dd>
      </dl><dl><dt>

          <code><a href="read_peak_file.html">read_peak_file()</a></code>

        </dt>
        <dd>Read MACS2/3 narrowPeak or SEACR BED peak file</dd>
      </dl><dl><dt>

          <code><a href="report_command.html">report_command()</a></code>

        </dt>
        <dd>Report command</dd>
      </dl><dl><dt>

          <code><a href="report_header.html">report_header()</a></code>

        </dt>
        <dd>Report header</dd>
      </dl><dl><dt>

          <code><a href="save_peak_file.html">save_peak_file()</a></code>

        </dt>
        <dd>Minimally save a peak object to a file (BED4)</dd>
      </dl><dl><dt>

          <code><a href="segregate_seqs.html">segregate_seqs()</a></code>

        </dt>
        <dd>Segregate input sequences into common and unique groups</dd>
      </dl><dl><dt>

          <code><a href="summit_to_motif.html">summit_to_motif()</a></code>

        </dt>
        <dd>Calculate the distance between peak summits and motifs</dd>
      </dl><dl><dt>

          <code><a href="to_plotly.html">to_plotly()</a></code>

        </dt>
        <dd>Convert ggplot2 objects to plotly</dd>
      </dl><dl><dt>

          <code><a href="trim_seqs.html">trim_seqs()</a></code>

        </dt>
        <dd>Trim sequences to a specified width around the summit</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hiranyamaya Dash, Thomas Roberts, Nathan Skene.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

