# Package index

## All functions

- [`CTCF_ChIP_peaks`](https://neurogenomics.github.io/MotifPeeker/reference/CTCF_ChIP_peaks.md)
  : Example ChIP-seq peak file
- [`CTCF_TIP_peaks`](https://neurogenomics.github.io/MotifPeeker/reference/CTCF_TIP_peaks.md)
  : Example TIP-seq peak file
- [`MotifPeeker()`](https://neurogenomics.github.io/MotifPeeker/reference/MotifPeeker.md)
  : Benchmark epigenomic profiling methods using motif enrichment
- [`bootstrap_distances()`](https://neurogenomics.github.io/MotifPeeker/reference/bootstrap_distances.md)
  : Bootstrap motif-summit distances for one set of peaks and one motif
- [`calc_frip()`](https://neurogenomics.github.io/MotifPeeker/reference/calc_frip.md)
  : Calculate FRiP score
- [`check_ENCODE()`](https://neurogenomics.github.io/MotifPeeker/reference/check_ENCODE.md)
  : Check for ENCODE input
- [`check_JASPAR()`](https://neurogenomics.github.io/MotifPeeker/reference/check_JASPAR.md)
  : Check for JASPAR input
- [`check_genome_build()`](https://neurogenomics.github.io/MotifPeeker/reference/check_genome_build.md)
  : Check genome build
- [`denovo_motifs()`](https://neurogenomics.github.io/MotifPeeker/reference/denovo_motifs.md)
  : Discover motifs in sequences
- [`find_motifs()`](https://neurogenomics.github.io/MotifPeeker/reference/find_motifs.md)
  : Find similar motifs
- [`get_JASPARCORE()`](https://neurogenomics.github.io/MotifPeeker/reference/get_JASPARCORE.md)
  : Download JASPAR CORE database
- [`get_df_distances()`](https://neurogenomics.github.io/MotifPeeker/reference/get_df_distances.md)
  : Get dataframe with motif-summit distances
- [`get_df_distances_bootstrapped()`](https://neurogenomics.github.io/MotifPeeker/reference/get_df_distances_bootstrapped.md)
  : Get dataframe with bootstrapped motif-summit distances
- [`get_df_enrichment()`](https://neurogenomics.github.io/MotifPeeker/reference/get_df_enrichment.md)
  : Get dataframe with motif enrichment values
- [`motif_MA1102.3`](https://neurogenomics.github.io/MotifPeeker/reference/motif_MA1102.3.md)
  : Example CTCFL JASPAR motif file
- [`motif_MA1930.2`](https://neurogenomics.github.io/MotifPeeker/reference/motif_MA1930.2.md)
  : Example CTCF JASPAR motif file
- [`motif_enrichment()`](https://neurogenomics.github.io/MotifPeeker/reference/motif_enrichment.md)
  : Calculate motif enrichment in a set of sequences
- [`motif_similarity()`](https://neurogenomics.github.io/MotifPeeker/reference/motif_similarity.md)
  : Compare motifs from segregated sequences
- [`read_motif_file()`](https://neurogenomics.github.io/MotifPeeker/reference/read_motif_file.md)
  : Read a motif file
- [`read_peak_file()`](https://neurogenomics.github.io/MotifPeeker/reference/read_peak_file.md)
  : Read MACS2/3 narrowPeak or SEACR BED peak file
- [`save_peak_file()`](https://neurogenomics.github.io/MotifPeeker/reference/save_peak_file.md)
  : Minimally save a peak object to a file (BED4)
- [`segregate_seqs()`](https://neurogenomics.github.io/MotifPeeker/reference/segregate_seqs.md)
  : Segregate input sequences into common and unique groups
- [`summit_to_motif()`](https://neurogenomics.github.io/MotifPeeker/reference/summit_to_motif.md)
  : Calculate the distance between peak summits and motifs
