<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Calculate the distance between peak summits and motifs — summit_to_motif • MotifPeeker</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate the distance between peak summits and motifs — summit_to_motif"><meta name="description" content="summit_to_motif() calculates the distance between each motif and its
nearest peak summit. runFimo from the memes package is used to
recover the locations of each motif."><meta property="og:description" content="summit_to_motif() calculates the distance between each motif and its
nearest peak summit. runFimo from the memes package is used to
recover the locations of each motif."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotifPeeker</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.12</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MotifPeeker.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/docker.html">Docker/Singularity Containers</a></li>
    <li><a class="dropdown-item" href="../articles/examples.html">Example Reports</a></li>
    <li><a class="dropdown-item" href="../articles/troubleshooting.html">Troubleshooting</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/neurogenomics/MotifPeeker/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Calculate the distance between peak summits and motifs</h1>
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MotifPeeker/blob/master/R/summit_to_motif.R" class="external-link"><code>R/summit_to_motif.R</code></a></small>
      <div class="d-none name"><code>summit_to_motif.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>summit_to_motif()</code> calculates the distance between each motif and its
nearest peak summit. <code>runFimo</code> from the <code>memes</code> package is used to
recover the locations of each motif.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">summit_to_motif</span><span class="op">(</span></span>
<span>  <span class="va">peak_input</span>,</span>
<span>  <span class="va">motif</span>,</span>
<span>  fp_rate <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  <span class="va">genome_build</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  meme_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-peak-input">peak_input<a class="anchor" aria-label="anchor" href="#arg-peak-input"></a></dt>
<dd><p>Either a path to the narrowPeak file or a GRanges peak
object generated by <code><a href="read_peak_file.html">read_peak_file()</a></code>.</p></dd>


<dt id="arg-motif">motif<a class="anchor" aria-label="anchor" href="#arg-motif"></a></dt>
<dd><p>An object of class <code>universalmotif</code>.</p></dd>


<dt id="arg-fp-rate">fp_rate<a class="anchor" aria-label="anchor" href="#arg-fp-rate"></a></dt>
<dd><p>The desired false-positive rate. A p-value threshold will be
selected based on this value. The default false-positive rate is
0.05.</p></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>The genome build that the peak sequences should be
derived from.</p></dd>


<dt id="arg-out-dir">out_dir<a class="anchor" aria-label="anchor" href="#arg-out-dir"></a></dt>
<dd><p>Location to save the 0-order background file. By default, the
background file will be written to a temporary directory.</p></dd>


<dt id="arg-meme-path">meme_path<a class="anchor" aria-label="anchor" href="#arg-meme-path"></a></dt>
<dd><p>path to "meme/bin/" (default: <code>NULL</code>). Will use default
search behavior as described in <code>check_meme_install()</code> if unset.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A logical indicating whether to print verbose messages while
running the function. (default = FALSE)</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/memes/man/runFimo.html" class="external-link">memes::runFimo</a></code></p><dl><dt><code>parse_genomic_coord</code></dt>
<dd><p><code>logical(1)</code> whether to parse genomic position
from fasta headers. Fasta headers must be UCSC format positions (ie
"chr:start-end"), but base 1 indexed (GRanges format). If names of fasta
entries are genomic coordinates and parse_genomic_coord == TRUE, results
will contain genomic coordinates of motif matches, otherwise FIMO will return
relative coordinates (i.e. positions from 1 to length of the fasta entry).</p></dd>

    <dt><code>skip_matched_sequence</code></dt>
<dd><p><code>logical(1)</code> whether or not to include the DNA
sequence of the match. Default: <code>FALSE</code>. Note: jobs will complete faster if
set to <code>TRUE</code>. <code>add_sequence()</code> can be used to lookup the sequence after data import if
<code>parse_genomic_coord</code> is <code>TRUE</code>, so setting this flag is not strictly needed.</p></dd>

    <dt><code>max_strand</code></dt>
<dd><p>if match is found on both strands, only report strand with
best match (default: TRUE).</p></dd>

    <dt><code>text</code></dt>
<dd><p><code>logical(1)</code> (default: <code>TRUE</code>). No output files will be created
on the filesystem. The results are unsorted and no q-values are computed.
This setting allows fast searches on very large inputs. When set to <code>FALSE</code>
FIMO will discard 50% of the lower significance matches if &gt;100,000 matches are
detected. <code>text = FALSE</code> will also incur a performance penalty because it
must first read a file to disk, then read it into memory. For these reasons,
I suggest keeping <code>text = TRUE</code>.</p></dd>

    <dt><code>silent</code></dt>
<dd><p><code>logical(1)</code> whether to suppress stdout/stderr printing to
console (default: TRUE). If the command is failing or giving unexpected
output, setting <code>silent = FALSE</code> can aid troubleshooting.</p></dd>


</dl></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing an expanded GRanges peak object with metadata
columns relating to motif positions along with a vector of summit-to-motif
distances for each valid peak.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>To calculate the p-value threshold for a desired false-positive rate, we use
the approximate formula:
$$p \approx \frac{fp\_rate}{2 \times \text{average peak width}}$$
(Dervied from
<a href="https://meme-suite.org/meme/doc/fimo-tutorial.html" class="external-link">FIMO documentation</a>)</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/memes/man/runAme.html" class="external-link">runAme</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu">memes</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/memes/man/meme_is_installed.html" class="external-link">meme_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CTCF_TIP_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"motif_MA1102.3"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">summit_to_motif</span><span class="op">(</span></span></span>
<span class="r-in"><span>    peak_input <span class="op">=</span> <span class="va">CTCF_TIP_peaks</span>,</span></span>
<span class="r-in"><span>    motif <span class="op">=</span> <span class="va">motif_MA1102.3</span>,</span></span>
<span class="r-in"><span>    fp_rate <span class="op">=</span> <span class="fl">5e-02</span>,</span></span>
<span class="r-in"><span>    genome_build <span class="op">=</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $peak_set</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 30 ranges and 10 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   seqnames            ranges strand |            name     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_15    chr10 67788408-67788724      * |  TIPseq_peak_15       129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_39    chr10 68988538-68988838      * |  TIPseq_peak_39       111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_42    chr10 69179798-69180156      * |  TIPseq_peak_42        75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_47    chr10 69408915-69409153      * |  TIPseq_peak_47       146</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_48    chr10 69413328-69413612      * |  TIPseq_peak_48       102</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...      ...               ...    ... .             ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_157    chr10 73529363-73529638      * | TIPseq_peak_157       106</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_162    chr10 73849014-73849393      * | TIPseq_peak_162       137</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_163    chr10 73879729-73880038      * | TIPseq_peak_163       248</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_167    chr10 73957451-73957745      * | TIPseq_peak_167        73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_170    chr10 73997853-73998166      * | TIPseq_peak_170        91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   signalValue    pValue    qValue      peak    summit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_15     9.06865   17.2614  12.96200       153  67788561</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_39     9.10857   15.2180  11.14710       168  68988706</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_42     7.28686   11.2099   7.52416       121  69179919</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_47     8.42133   19.1676  14.67840       110  69409025</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_48     7.00187   14.2174  10.25130       141  69413469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...         ...       ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_157     8.79456   14.6750  10.66540       105  73529468</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_162     9.57946   18.1045  13.72430       240  73849254</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_163    12.53940   30.6089  24.84330       140  73879869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_167     6.58212   11.0334   7.35935       125  73957576</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_170     7.78776   12.9930   9.14808       138  73997991</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   motif_start motif_end distance_to_summit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;integer&gt; &lt;integer&gt;          &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_15    67788624  67788631              -66.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_39    68988582  68988589              120.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_42    69179983  69179990              -67.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_47    69408938  69408945               83.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    TIPseq_peak_48    69413548  69413555              -82.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...         ...       ...                ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_157    73529430  73529437               34.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_162    73849208  73849215               42.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_163    73879829  73879836               36.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_167    73957554  73957561               18.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TIPseq_peak_170    73998063  73998070              -75.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $distance_to_summit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1]  -66.5  120.5  -67.5   83.5  -82.5  -48.5   25.5   50.5  -59.5  -19.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11]  -70.5 -110.5 -316.5  387.5  -45.5  -64.5   91.5  -60.5   48.5  -26.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [21]   31.5 -107.5 -189.5 -146.5 -114.5   34.5   42.5   36.5   18.5  -75.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hiranyamaya Dash, Thomas Roberts, Maria Weinert, Nathan Skene.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

