<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Example Reports • MotifPeeker</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Example Reports">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotifPeeker</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/MotifPeeker.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/docker.html">Docker/Singularity Containers</a></li>
    <li><a class="dropdown-item" href="../articles/examples.html">Example Reports</a></li>
    <li><a class="dropdown-item" href="../articles/troubleshooting.html">Troubleshooting</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/neurogenomics/MotifPeeker/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Example Reports</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MotifPeeker/blob/revision_changes/vignettes/articles/examples.Rmd" class="external-link"><code>vignettes/articles/examples.Rmd</code></a></small>
      <div class="d-none name"><code>examples.Rmd</code></div>
    </div>

    
    
<p><strong>Updated:</strong> <strong><em>Nov-20-2025</em></strong></p>
<p><a href="https://github.com/neurogenomics/MotifPeeker-reports/" class="external-link"><code>MotifPeeker-reports</code></a>
is a public archive for interactive HTML reports generated by <a href="https://github.com/neurogenomics/MotifPeeker" class="external-link"><code>MotifPeeker</code></a>,
along with the code used to produce them.</p>
<div class="section level3">
<h3 id="reports-home">
<a href="https://neurogenomics.github.io/MotifPeeker-reports/">Reports
Home</a><a class="anchor" aria-label="anchor" href="#reports-home"></a>
</h3>
<p><br></p>
</div>
<div class="section level3">
<h3 id="reports">Reports<a class="anchor" aria-label="anchor" href="#reports"></a>
</h3>
<div class="section level4">
<h4 id="ctcf-chip-seq---tip-seq">
<a href="https://neurogenomics.github.io/MotifPeeker-reports/CTCF_ChIP-TIP/output/MotifPeeker_20251106_130040/MotifPeeker.html">CTCF
ChIP-seq - TIP-seq</a><a class="anchor" aria-label="anchor" href="#ctcf-chip-seq---tip-seq"></a>
</h4>
<details><summary>
Dataset Details
</summary><p>Compares three public ChIP-seq datasets against a TIP-seq dataset for
CTCF in HCT116 cells. Details of the datasets used are as follows:</p>
<ol style="list-style-type: decimal">
<li>
<code>ChIPSeq1</code><br>
ENCODE Accession: <a href="https://www.encodeproject.org/files/ENCFF172XLC/" class="external-link">ENCFF172XLC</a><br>
</li>
<li>
<code>ChIPSeq2</code><br>
ENCODE Accession: <a href="https://www.encodeproject.org/files/ENCFF411OOS/" class="external-link">ENCFF411OOS</a><br>
</li>
<li>
<code>ChIPSeq3</code><br>
ENCODE Accession: <a href="https://www.encodeproject.org/files/ENCFF091ODJ/" class="external-link">ENCFF091ODJ</a>
</li>
<li>
<code>TIPSeq</code><br>
Alignment file produced by running the <a href="https://nf-co.re/cutandrun/" class="external-link"><code>nf-core/cutandrun</code></a>
pipeline on the raw sequencing data sourced from <em>NIH Sequence Read
Archives</em> <a href="https://trace.ncbi.nlm.nih.gov/Traces/index.html?view=run_browser&amp;acc=SRR16963166&amp;display=metadata" class="external-link"><em>ID:
SRR16963166</em></a>.</li>
</ol>
<p>Peak calling was performed using <a href="https://github.com/macs3-project/MACS" class="external-link"><code>MACS3</code></a>
using the following commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## For ChIP-Seq datasets</span></span>
<span><span class="fu">MACSr</span><span class="fu">::</span><span class="fu">callpeak</span><span class="op">(</span></span>
<span>        tfile <span class="op">=</span> <span class="va">...</span>,</span>
<span>        cfile <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># No control file</span></span>
<span>        nomodel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        qvalue <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>        format <span class="op">=</span> <span class="st">"BAM"</span>,</span>
<span>        name <span class="op">=</span> <span class="st">"chipseq"</span>,</span>
<span>        outdir <span class="op">=</span> <span class="va">...</span></span>
<span>    <span class="op">)</span></span>
<span>    </span>
<span><span class="co">## For TIP-Seq dataset</span></span>
<span><span class="fu">MACSr</span><span class="fu">::</span><span class="fu">callpeak</span><span class="op">(</span></span>
<span>        tfile <span class="op">=</span> <span class="va">tipseq_alignment_data</span>,</span>
<span>        nomodel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        qvalue <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>        shift <span class="op">=</span> <span class="op">-</span><span class="fl">75</span>,  <span class="co"># 5' shift</span></span>
<span>        extsize <span class="op">=</span> <span class="fl">150</span>,</span>
<span>        keepduplicates <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>        format <span class="op">=</span> <span class="st">"BAM"</span>,</span>
<span>        name <span class="op">=</span> <span class="st">"tipseq"</span>,</span>
<span>        outdir <span class="op">=</span> <span class="va">...</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Motif files used:</p>
<ol style="list-style-type: decimal">
<li>CTCF motif <a href="https://jaspar.elixir.no/matrix/MA1930.2/" class="external-link">MA1930.2</a>
</li>
<li>CTCFL motif <a href="https://jaspar.elixir.no/matrix/MA1102.3/" class="external-link">MA1102.3</a>
</li>
</ol></details><p> <a href="https://github.com/neurogenomics/MotifPeeker-reports/blob/master/CTCF_ChIP-TIP/run.R" class="external-link">Code</a></p>
</div>
<div class="section level4">
<h4 id="built-in-data">
<a href="https://neurogenomics.github.io/MotifPeeker-reports/built_in_data/output/MotifPeeker_20251120_120202/MotifPeeker.html">Built-in
data</a><a class="anchor" aria-label="anchor" href="#built-in-data"></a>
</h4>
<details><summary>
Dataset Details
</summary>
Report generated from built-in data, as described in the
<a href="https://neurogenomics.github.io/MotifPeeker/articles/MotifPeeker.html">
<code>MotifPeeker</code> vignette</a>.
</details><p> <a href="https://github.com/neurogenomics/MotifPeeker-reports/blob/master/built_in_data/run.R" class="external-link">Code</a></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] digest_0.6.38     desc_1.4.3        R6_2.6.1          fastmap_1.2.0    </span></span>
<span><span class="co">##  [5] xfun_0.54         cachem_1.1.0      knitr_1.50        htmltools_0.5.8.1</span></span>
<span><span class="co">##  [9] rmarkdown_2.30    lifecycle_1.0.4   cli_3.6.5         sass_0.4.10      </span></span>
<span><span class="co">## [13] pkgdown_2.2.0     textshaping_1.0.1 jquerylib_0.1.4   systemfonts_1.2.3</span></span>
<span><span class="co">## [17] compiler_4.5.1    tools_4.5.1       ragg_1.4.0        bslib_0.9.0      </span></span>
<span><span class="co">## [21] evaluate_1.0.5    yaml_2.3.10       jsonlite_2.0.0    rlang_1.1.6      </span></span>
<span><span class="co">## [25] fs_1.6.6          htmlwidgets_1.6.4</span></span></code></pre>
</details><p><br></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hiranyamaya Dash, Thomas Roberts, Maria Weinert, Nathan Skene.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
