# Example Reports

**Updated:** ***Nov-20-2025***

[`MotifPeeker-reports`](https://github.com/neurogenomics/MotifPeeker-reports/)
is a public archive for interactive HTML reports generated by
[`MotifPeeker`](https://github.com/neurogenomics/MotifPeeker), along
with the code used to produce them.

### [Reports Home](https://neurogenomics.github.io/MotifPeeker-reports/)

  

### Reports

#### [CTCF ChIP-seq - TIP-seq](https://neurogenomics.github.io/MotifPeeker-reports/CTCF_ChIP-TIP/output/MotifPeeker_20251106_130040/MotifPeeker.html)

Dataset Details

Compares three public ChIP-seq datasets against a TIP-seq dataset for
CTCF in HCT116 cells. Details of the datasets used are as follows:

1.  `ChIPSeq1`  
    ENCODE Accession:
    [ENCFF172XLC](https://www.encodeproject.org/files/ENCFF172XLC/)  
2.  `ChIPSeq2`  
    ENCODE Accession:
    [ENCFF411OOS](https://www.encodeproject.org/files/ENCFF411OOS/)  
3.  `ChIPSeq3`  
    ENCODE Accession:
    [ENCFF091ODJ](https://www.encodeproject.org/files/ENCFF091ODJ/)
4.  `TIPSeq`  
    Alignment file produced by running the
    [`nf-core/cutandrun`](https://nf-co.re/cutandrun/) pipeline on the
    raw sequencing data sourced from *NIH Sequence Read Archives* [*ID:
    SRR16963166*](https://trace.ncbi.nlm.nih.gov/Traces/index.html?view=run_browser&acc=SRR16963166&display=metadata).

Peak calling was performed using
[`MACS3`](https://github.com/macs3-project/MACS) using the following
commands:

``` r
## For ChIP-Seq datasets
MACSr::callpeak(
        tfile = ...,
        cfile = NULL,  # No control file
        nomodel = FALSE,
        qvalue = 0.01,
        format = "BAM",
        name = "chipseq",
        outdir = ...
    )
    
## For TIP-Seq dataset
MACSr::callpeak(
        tfile = tipseq_alignment_data,
        nomodel = TRUE,
        qvalue = 0.01,
        shift = -75,  # 5' shift
        extsize = 150,
        keepduplicates = "all",
        format = "BAM",
        name = "tipseq",
        outdir = ...
    )
```

Motif files used:

1.  CTCF motif [MA1930.2](https://jaspar.elixir.no/matrix/MA1930.2/)
2.  CTCFL motif [MA1102.3](https://jaspar.elixir.no/matrix/MA1102.3/)

 [Code](https://github.com/neurogenomics/MotifPeeker-reports/blob/master/CTCF_ChIP-TIP/run.R)

#### [Built-in data](https://neurogenomics.github.io/MotifPeeker-reports/built_in_data/output/MotifPeeker_20251120_120202/MotifPeeker.html)

Dataset Details

Report generated from built-in data, as described in the [`MotifPeeker`
vignette](https://neurogenomics.github.io/MotifPeeker/articles/MotifPeeker.html).

 [Code](https://github.com/neurogenomics/MotifPeeker-reports/blob/master/built_in_data/run.R)

## Session Info

``` r
utils::sessionInfo()
```

    ## R version 4.5.1 (2025-06-13)
    ## Platform: x86_64-pc-linux-gnu
    ## Running under: Ubuntu 24.04.2 LTS
    ## 
    ## Matrix products: default
    ## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
    ## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0
    ## 
    ## locale:
    ##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
    ##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
    ##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
    ##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
    ##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
    ## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
    ## 
    ## time zone: UTC
    ## tzcode source: system (glibc)
    ## 
    ## attached base packages:
    ## [1] stats     graphics  grDevices utils     datasets  methods   base     
    ## 
    ## loaded via a namespace (and not attached):
    ##  [1] digest_0.6.38     desc_1.4.3        R6_2.6.1          fastmap_1.2.0    
    ##  [5] xfun_0.54         cachem_1.1.0      knitr_1.50        htmltools_0.5.8.1
    ##  [9] rmarkdown_2.30    lifecycle_1.0.4   cli_3.6.5         sass_0.4.10      
    ## [13] pkgdown_2.2.0     textshaping_1.0.1 jquerylib_0.1.4   systemfonts_1.2.3
    ## [17] compiler_4.5.1    tools_4.5.1       ragg_1.4.0        bslib_0.9.0      
    ## [21] evaluate_1.0.5    yaml_2.3.10       jsonlite_2.0.0    rlang_1.1.6      
    ## [25] fs_1.6.6          htmlwidgets_1.6.4

  
