<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Get started • MotifPeeker</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotifPeeker</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/MotifPeeker.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/docker.html">Docker/Singularity Containers</a></li>
    <li><a class="dropdown-item" href="../articles/examples.html">Example Reports</a></li>
    <li><a class="dropdown-item" href="../articles/troubleshooting.html">Troubleshooting</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/neurogenomics/MotifPeeker/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get started</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MotifPeeker/blob/master/vignettes/MotifPeeker.Rmd" class="external-link"><code>vignettes/MotifPeeker.Rmd</code></a></small>
      <div class="d-none name"><code>MotifPeeker.Rmd</code></div>
    </div>

    
    
<p><strong>Updated:</strong> <strong><em>May-29-2025</em></strong></p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em>MotifPeeker</em> package facilitates the comparison and
validation of datasets from epigenomic profiling methods, using motif
enrichment as the key benchmark. The package generates a comprehensive
summary report with results from various downstream analyses by
processing peak, alignment, and motif files. This allows for detailed
statistical analysis of multiple epigenomic datasets without any coding,
ensuring both accessibility and robustness.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The rapidly advancing field of epigenomics has led to the development
of various techniques for profiling protein interactions with DNA,
enhancing our understanding of gene regulatory mechanisms and genetic
factors behind complex diseases. However, the validation of these newer
methods, such as CUT&amp;RUN, CUT&amp;TAG and TIP-Seq, remains a
critical area that requires further exploration, especially given their
potential to address the challenges of traditional ChIP-Seq.</p>
<p>Common epigenomic profiling techniques rely on target proteins, such
as the transcriptional regulator CTCF, binding to their respective sites
on the DNA to isolate the sequences for sequencing. These binding sites
may contain specific sequences recognised by the transcription factors,
called motifs. Unlike other comparison tools like <em>ChIPseeker</em>
and <em>EpiCompare</em>, <em>MotifPeeker</em> checks for the presence of
these motifs in the sequences enriched from epigenomic profiling methods
as a novel strategy to benchmark them.</p>
<p>At the same time, general metrics like FRiP scores and peak width
distributions are also reported to add more context to the comparisons.
While the goal remains to benchmark different epigenomic datasets,
<em>MotifPeeker</em> can also be used to compare the effects of various
downstream processing, such as the thresholds for peak calling and the
choice of the peak caller itself. The package can also help identify
differences arising from different experimental conditions or protocol
optimisations.</p>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p><em>MotifPeeker</em> comes with a small subset of two epigenomic
datasets targeting CTCF in HCT116 cells, generated using ChIP-Seq and
TIP-Seq.</p>
<ul>
<li>ChIP-Seq alignment file (<code>CTCF_ChIP_alignment.bam</code>)
sourced from the ENCODE project (<a href="https://www.encodeproject.org/files/ENCFF091ODJ/" class="external-link">Accession:
ENCFF091ODJ</a>).<br>
</li>
<li>TIP-Seq alignment file (<code>CTCF_TIP_alignment.bam</code>) was
manually processed using the <a href="https://nf-co.re/cutandrun/3.2.2" class="external-link"><code>nf-core/cutandrun</code></a>
pipeline. The raw read files were sourced from <em>NIH Sequence Read
Archives</em> (<a href="https://trace.ncbi.nlm.nih.gov/Traces/index.html?view=run_browser&amp;acc=SRR16963166" class="external-link"><em>ID:
SRR16963166</em></a>).</li>
</ul>
<p>The alignment files were processed using the <em>MACS3</em> peak
caller to produce their respective peak files with the
<code>q-value</code> parameter set to 0.01.</p>
<p>Two motif files for CTCF are also bundled with the package:</p>
<ul>
<li>JASPAR motif file - <a href="https://jaspar.elixir.no/matrix/MA1930.2/" class="external-link">MA1930.2</a><br>
</li>
<li>JASPAR motif file - <a href="https://jaspar.elixir.no/matrix/MA1102.3/" class="external-link">MA1102.3</a>
</li>
</ul>
<p>Please note that the peaks and alignments included are a very small
subset (<em>chr10:65,654,529-74,841,155</em>) of the actual data. It
only serves as an example to demonstrate the package and run tests to
maintain the integrity of the package.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>MotifPeeker</code> uses <a href="https://www.bioconductor.org/packages/release/bioc/html/memes.html" class="external-link"><code>memes</code></a>
which relies on a local install of the <a href="https://meme-suite.org/meme/" class="external-link">MEME suite</a>, which can be
installed as follows:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="va">MEME_VERSION</span><span class="op">=</span>5.5.5  <span class="co"># or the latest version</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">wget</span> https://meme-suite.org/meme/meme-software/<span class="va">$MEME_VERSION</span>/meme-<span class="va">$MEME_VERSION</span>.tar.gz</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">tar</span> zxf meme-<span class="va">$MEME_VERSION</span>.tar.gz</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="bu">cd</span> meme-<span class="va">$MEME_VERSION</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">./configure</span> <span class="at">--prefix</span><span class="op">=</span><span class="va">$HOME</span>/meme <span class="at">--with-url</span><span class="op">=</span>http://meme-suite.org/ <span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>--enable-build-libxml2 <span class="at">--enable-build-libxslt</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">make</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">make</span> install</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co"># Add to PATH</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export PATH=$HOME/meme/bin:$HOME/meme/libexec/meme-$MEME_VERSION:$PATH'</span> <span class="op">&gt;&gt;</span> ~/.bashrc</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export MEME_BIN=$HOME/meme/bin'</span> <span class="op">&gt;&gt;</span> ~/.bashrc</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="bu">source</span> ~/.bashrc</span></code></pre></div>
<p><strong>NOTE:</strong> It is important that Perl dependencies
associated with MEME suite are also installed, particularly
<code>XML::Parser</code>, which can be installed using the following
command in the terminal:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">cpan</span> install XML::Parser</span></code></pre></div>
<p>For more information, refer to the <a href="https://meme-suite.org/meme/doc/install.html#prereq_perl" class="external-link">Perl
dependency section of the MEME suite</a>.</p>
<p>Once the MEME suite and its associated Perl dependencies are
installed, install and load <code>MotifPeeker</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install latest version of MotifPeeker</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MotifPeeker"</span>, version <span class="op">=</span> <span class="st">"devel"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/MotifPeeker" class="external-link">MotifPeeker</a></span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, you can use the <a href="https://neurogenomics.github.io/MotifPeeker/articles/docker.html">Docker/Singularity
container</a> to run the package out-of-the-box.</p>
</div>
<div class="section level2">
<h2 id="running-motifpeeker">Running <em>MotifPeeker</em><a class="anchor" aria-label="anchor" href="#running-motifpeeker"></a>
</h2>
<p>In this example, we will compare the bundled ChIP-Seq dataset against
the TIP-Seq dataset.</p>
<div class="section level3">
<h3 id="load-the-package">Load the package<a class="anchor" aria-label="anchor" href="#load-the-package"></a>
</h3>
<p>Once installed, load the package using:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/MotifPeeker" class="external-link">MotifPeeker</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-the-example-datasets">Load the example datasets<a class="anchor" aria-label="anchor" href="#load-the-example-datasets"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Peak files processed using read_peak_file()</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CTCF_ChIP_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CTCF_TIP_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Motif files processed using read_motif_file()</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"motif_MA1102.3"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"motif_MA1930.2"</span>, package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-input-data">Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h3>
<div class="section level4">
<h4 id="peak-files">Peak Files<a class="anchor" aria-label="anchor" href="#peak-files"></a>
</h4>
<p><em>MotifPeeker</em> accepts lists of both <code>GRanges</code>
objects produced by <code><a href="../reference/read_peak_file.html">read_peak_file()</a></code>, or paths to the
<em>MACS2/3</em> <code>.narrowPeak</code> files or <em>SEACR</em>
<code>.bed</code> files, or ENCODE file IDs to automatically download
the respective files.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## MACS2/3 peak files</span></span>
<span><span class="va">peak_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"/path/to/peak1.narrowPeak"</span>, <span class="st">"/path/to/peak2.narrowPeak"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## or SEACR peak files</span></span>
<span><span class="va">peak_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"/path/to/peak1.bed"</span>, <span class="st">"/path/to/peak2.bed"</span><span class="op">)</span></span></code></pre></div>
<p>In this example, we will use the bundled <code>GRanges</code>
peaks:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peak_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CTCF_ChIP_peaks</span>, <span class="va">CTCF_TIP_peaks</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="alignment-files">Alignment Files<a class="anchor" aria-label="anchor" href="#alignment-files"></a>
</h4>
<p>Optionally provide a list of path to <code>.bam</code> alignment
files, or ENCODE file IDs to generate additional comparisons like FRiP
scores.</p>
<p>In this example, we will use the built-in alignment files.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Alignment files</span></span>
<span><span class="va">CTCF_ChIP_alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"CTCF_ChIP_alignment.bam"</span>,</span>
<span>                                    package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span>
<span><span class="va">CTCF_TIP_alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"CTCF_TIP_alignment.bam"</span>,</span>
<span>                                    package <span class="op">=</span> <span class="st">"MotifPeeker"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alignment_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CTCF_ChIP_alignment</span>, <span class="va">CTCF_TIP_alignment</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="motif-files">Motif Files<a class="anchor" aria-label="anchor" href="#motif-files"></a>
</h4>
<p><em>MotifPeeker</em> accepts a list of either
<code>universalmotif</code> objects, or paths to the
<code>.jaspar</code> files.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## JASPAR motif files</span></span>
<span><span class="va">motif_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"/path/to/motif1.jaspar"</span>, <span class="st">"/path/to/motif2.jaspar"</span><span class="op">)</span></span></code></pre></div>
<p>If you use JASPAR motif files, it is recommended that you label them
by using the <code>motif_labels</code> parameter of the
<code><a href="../reference/MotifPeeker.html">MotifPeeker()</a></code> function.</p>
<p>In this example, we will use the bundled <code>universalmotif</code>
motifs:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motif_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">motif_MA1102.3</span>, <span class="va">motif_MA1930.2</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="run-motifpeeker">Run <em>MotifPeeker</em><a class="anchor" aria-label="anchor" href="#run-motifpeeker"></a>
</h3>
<p>The report can be generated by using the main function
<code><a href="../reference/MotifPeeker.html">MotifPeeker()</a></code>. For more run customisations, refer to the
next sections.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu">MotifPeeker</span><span class="fu">:::</span><span class="fu"><a href="../reference/confirm_meme_install.html">confirm_meme_install</a></span><span class="op">(</span>continue <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/MotifPeeker.html">MotifPeeker</a></span><span class="op">(</span></span>
<span>        peak_files <span class="op">=</span> <span class="va">peak_files</span>,</span>
<span>        reference_index <span class="op">=</span> <span class="fl">2</span>,  <span class="co"># Set TIP-seq experiment as reference</span></span>
<span>        alignment_files <span class="op">=</span> <span class="va">alignment_files</span>,</span>
<span>        exp_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ChIP"</span>, <span class="st">"TIP"</span><span class="op">)</span>,</span>
<span>        exp_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chipseq"</span>, <span class="st">"tipseq"</span><span class="op">)</span>,</span>
<span>        genome_build <span class="op">=</span> <span class="st">"hg38"</span>,  <span class="co"># Use hg38 genome build</span></span>
<span>        motif_files <span class="op">=</span> <span class="va">motif_files</span>,</span>
<span>        cell_counts <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># No cell-count information</span></span>
<span>        motif_discovery <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        motif_discovery_count <span class="op">=</span> <span class="fl">3</span>,  <span class="co"># Discover top 3 motifs</span></span>
<span>        motif_db <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># Use default motif database (JASPAR)</span></span>
<span>        download_buttons <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,  <span class="co"># Save output in a temporary directory</span></span>
<span>        BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>,  <span class="co"># Use two CPU cores on a 16GB RAM machine</span></span>
<span>        debug <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        quiet <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## Starting run with 1 cores.</span></span></code></pre>
<pre><code><span><span class="co">## Script run successfully. </span></span>
<span><span class="co">## Output saved at: /tmp/Rtmpj15ZhI/MotifPeeker_20250529_215642 </span></span>
<span><span class="co">## Time taken: 1.05 mins.</span></span></code></pre>
<pre><code><span><span class="co">## [1] "/tmp/Rtmpj15ZhI/MotifPeeker_20250529_215642"</span></span></code></pre>
<div class="section level4">
<h4 id="required-inputs">Required Inputs<a class="anchor" aria-label="anchor" href="#required-inputs"></a>
</h4>
<p>These input parameters must be provided:</p>
<details><summary><strong>Details</strong>
</summary><ul>
<li>
<code>peak_files</code>: A list of path to peak files or
<code>GRanges</code> objects with the peaks to analyse. Currently, only
peak files from <code>MACS2/3</code> (<code>.narrowPeak</code>) and
<code>SEACR</code> (<code>.bed</code>) are supported. ENCODE file IDs
can also be provided to automatically fetch peak file(s) from the ENCODE
database.<br>
</li>
<li>
<code>reference_index</code>: An integer specifying the index of the
reference dataset in the <code>peak_files</code> list to use as
reference for various comparisons. (default = 1)<br>
</li>
<li>
<code>genome_build</code>: A character string or a
<code>BSgenome</code> object specifying the genome build of the
datasets. At the moment, only hg38 and hg19 are supported as abbreviated
input.<br>
</li>
<li>
<code>out_dir</code>: A character string specifying the output
directory to save the HTML report and other files.</li>
</ul></details>
</div>
<div class="section level4">
<h4 id="optional-inputs">Optional Inputs<a class="anchor" aria-label="anchor" href="#optional-inputs"></a>
</h4>
<p>These input parameters optional, but <em>recommended</em> to add more
analyses, or enhance them:</p>
<details><summary><strong>Details</strong>
</summary><ul>
<li>
<code>alignment_files</code>: A list of path to alignment files or
<code><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html" class="external-link">Rsamtools::BamFile</a></code> objects with the alignment sequences to
analyse. Alignment files are used to calculate read-related metrics like
FRiP score. ENCODE file IDs can also be provided to automatically fetch
alignment file(s) from the ENCODE database.<br>
</li>
<li>
<code>exp_labels</code>: A character vector of labels for each peak
file. If not provided, capital letters will be used as labels in the
report.</li>
<li>
<code>exp_type</code>: A character vector of experimental types for
each peak file.<br>
Useful for comparison of different methods. If not provided, all
datasets will be classified as “unknown” experiment types in the report.
<code>exp_type</code> is used only for labelling. It does not affect the
analyses. You can also input custom strings. Datasets will be grouped as
long as they match their respective <code>exp_type</code>. Supported
experimental types are:<br>
- <code>chipseq</code>: ChIP-seq data<br>
- <code>tipseq</code>: TIP-seq data<br>
- <code>cuttag</code>: CUT&amp;Tag data<br>
- <code>cutrun</code>: CUT&amp;Run data<br>
</li>
<li>
<code>motif_files</code>: A character vector of path to motif files,
or a vector of <code>universalmotif-class</code> objects. Required to
run Known Motif Enrichment Analysis. JASPAR matrix IDs can also be
provided to automatically fetch motifs from the JASPAR.<br>
</li>
<li>
<code>motif_labels</code>: A character vector of labels for each
motif file. Only used if path to file names are passed in motif_files.
If not provided, the motif file names will be used as labels.<br>
</li>
<li>
<code>cell_counts</code>: An integer vector of experiment cell
counts for each peak file (if available). Creates additional comparisons
based on cell counts.<br>
</li>
<li>
<code>motif_db</code>: Path to <code>.meme</code> format file to use
as reference database, or a list of <code>universalmotif-class</code>
objects. Results from motif discovery are searched against this database
to find similar motifs. If not provided, JASPAR CORE database will be
used, making this parameter <strong>truly optional</strong>.
<strong>NOTE</strong>: p-value estimates are inaccurate when the
database has fewer than 50 entries.</li>
</ul></details>
</div>
<div class="section level4">
<h4 id="other-options">Other Options<a class="anchor" aria-label="anchor" href="#other-options"></a>
</h4>
<p>For more information on additional parameters, please refer to the
documentation for <a href="https://neurogenomics.github.io/MotifPeeker/reference/MotifPeeker.html"><code>MotifPeeker()</code></a>.</p>
</div>
<div class="section level4">
<h4 id="runtime">Runtime Guidance<a class="anchor" aria-label="anchor" href="#runtime"></a>
</h4>
<p>For 4 datasets, the runtime is approximately 3 minutes with
motif_discovery disabled. However, motif discovery can take hours to
complete.</p>
<p>To make computation faster, we highly recommend tuning the following
arguments:</p>
<details><summary><strong>Details</strong>
</summary><ul>
<li>
<code>BPPARAM = Multicore(x)</code>: Running motif discovery in
parallel can significantly reduce runtime, but it is very
memory-intensive, consuming upwards of 10GB of RAM per thread. Memory
starvation can greatly slow the process, so set workers (x) with
caution.<br>
</li>
<li>
<code>motif_discovery_count</code>: The number of motifs to discover
per sequence group exponentially increases runtime. We recommend no more
than 5 motifs to make a meaningful inference.<br>
</li>
<li>
<code>trim_seq_width</code>: Trimming sequences before running motif
discovery can significantly reduce the search space. Sequence length can
exponentially increase runtime. We recommend running the script with
<code>motif_discovery = FALSE</code> and studying the motif-summit
distance distribution under general metrics to find the sequence length
that captures most motifs. A good starting point is 150 but it can be
reduced further if appropriate.</li>
</ul></details>
</div>
</div>
<div class="section level3">
<h3 id="outputs">Outputs<a class="anchor" aria-label="anchor" href="#outputs"></a>
</h3>
<p><code>MotifPeeker</code> generates its output in a new folder within
he <code>out_dir</code> directory. The folder is named
<code>MotifPeeker_YYYYMMDD_HHMMSS</code> and contains the following
files:</p>
<ul>
<li>
<code>MotifPeeker.html</code>: The main HTML report, including all
analyses and plots.<br>
</li>
<li>Output from various MEME suite tools in their respecive
sub-directories, if <code>save_runfiles</code> is set to
<code>TRUE</code>.</li>
</ul>
</div>
<div class="section level3">
<h3 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h3>
<p>If something does not work as expected, refer to <a href="https://neurogenomics.github.io/MotifPeeker/articles/troubleshooting.html">troubleshooting</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="future-enhancements">Future Enhancements<a class="anchor" aria-label="anchor" href="#future-enhancements"></a>
</h2>
<ul>
<li>Add support for outputs from more peak callers.<br>
</li>
<li>Automatically detect ideal <code>trim_peak_width</code> to reduce
motif discovery runtime.<br>
</li>
<li>Add more <a href="https://neurogenomics.github.io/MotifPeeker/articles/troubleshooting.html">troubleshooting</a>
steps to the documentation.</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] MotifPeeker_1.1.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3                jsonlite_2.0.0                   </span></span>
<span><span class="co">##   [3] magrittr_2.0.3                    farver_2.1.2                     </span></span>
<span><span class="co">##   [5] rmarkdown_2.29                    fs_1.6.6                         </span></span>
<span><span class="co">##   [7] BiocIO_1.19.0                     ragg_1.4.0                       </span></span>
<span><span class="co">##   [9] vctrs_0.6.5                       memoise_2.0.1                    </span></span>
<span><span class="co">##  [11] Rsamtools_2.25.0                  b64_0.1.6                        </span></span>
<span><span class="co">##  [13] RCurl_1.98-1.17                   webshot_0.5.5                    </span></span>
<span><span class="co">##  [15] htmltools_0.5.8.1                 S4Arrays_1.9.0                   </span></span>
<span><span class="co">##  [17] curl_6.2.3                        SparseArray_1.9.0                </span></span>
<span><span class="co">##  [19] sass_0.4.10                       bslib_0.9.0                      </span></span>
<span><span class="co">##  [21] htmlwidgets_1.6.4                 desc_1.4.3                       </span></span>
<span><span class="co">##  [23] plyr_1.8.9                        testthat_3.2.3                   </span></span>
<span><span class="co">##  [25] httr2_1.1.2                       lubridate_1.9.4                  </span></span>
<span><span class="co">##  [27] plotly_4.10.4                     cachem_1.1.0                     </span></span>
<span><span class="co">##  [29] GenomicAlignments_1.45.0          mime_0.13                        </span></span>
<span><span class="co">##  [31] downloadthis_0.4.1                lifecycle_1.0.4                  </span></span>
<span><span class="co">##  [33] iterators_1.0.14                  pkgconfig_2.0.3                  </span></span>
<span><span class="co">##  [35] Matrix_1.7-3                      R6_2.6.1                         </span></span>
<span><span class="co">##  [37] fastmap_1.2.0                     MatrixGenerics_1.21.0            </span></span>
<span><span class="co">##  [39] digest_0.6.37                     S4Vectors_0.47.0                 </span></span>
<span><span class="co">##  [41] ps_1.9.1                          rprojroot_2.0.4                  </span></span>
<span><span class="co">##  [43] pkgload_1.4.0                     crosstalk_1.2.1                  </span></span>
<span><span class="co">##  [45] textshaping_1.0.1                 GenomicRanges_1.61.0             </span></span>
<span><span class="co">##  [47] RSQLite_2.3.11                    seriation_1.5.7                  </span></span>
<span><span class="co">##  [49] bsplus_0.1.5                      labeling_0.4.3                   </span></span>
<span><span class="co">##  [51] filelock_1.0.3                    timechange_0.3.0                 </span></span>
<span><span class="co">##  [53] httr_1.4.7                        abind_1.4-8                      </span></span>
<span><span class="co">##  [55] compiler_4.5.0                    bit64_4.6.0-1                    </span></span>
<span><span class="co">##  [57] withr_3.0.2                       BiocParallel_1.43.2              </span></span>
<span><span class="co">##  [59] viridis_0.6.5                     DBI_1.2.3                        </span></span>
<span><span class="co">##  [61] heatmaply_1.5.0                   dendextend_1.19.0                </span></span>
<span><span class="co">##  [63] R.utils_2.13.0                    MASS_7.3-65                      </span></span>
<span><span class="co">##  [65] rappdirs_0.3.3                    DelayedArray_0.35.1              </span></span>
<span><span class="co">##  [67] rjson_0.2.23                      tools_4.5.0                      </span></span>
<span><span class="co">##  [69] zip_2.3.3                         ggseqlogo_0.2                    </span></span>
<span><span class="co">##  [71] R.oo_1.27.1                       glue_1.8.0                       </span></span>
<span><span class="co">##  [73] restfulr_0.0.15                   grid_4.5.0                       </span></span>
<span><span class="co">##  [75] reshape2_1.4.4                    generics_0.1.4                   </span></span>
<span><span class="co">##  [77] gtable_0.3.6                      BSgenome_1.77.0                  </span></span>
<span><span class="co">##  [79] tzdb_0.5.0                        R.methodsS3_1.8.2                </span></span>
<span><span class="co">##  [81] ca_0.71.1                         tidyr_1.3.1                      </span></span>
<span><span class="co">##  [83] data.table_1.17.4                 hms_1.1.3                        </span></span>
<span><span class="co">##  [85] xml2_1.3.8                        XVector_0.49.0                   </span></span>
<span><span class="co">##  [87] BiocGenerics_0.55.0               memes_1.17.0                     </span></span>
<span><span class="co">##  [89] foreach_1.5.2                     pillar_1.10.2                    </span></span>
<span><span class="co">##  [91] stringr_1.5.1                     vroom_1.6.5                      </span></span>
<span><span class="co">##  [93] dplyr_1.1.4                       BiocFileCache_2.99.5             </span></span>
<span><span class="co">##  [95] lattice_0.22-7                    rtracklayer_1.69.0               </span></span>
<span><span class="co">##  [97] bit_4.6.0                         universalmotif_1.27.2            </span></span>
<span><span class="co">##  [99] tidyselect_1.2.1                  BSgenome.Hsapiens.UCSC.hg38_1.4.5</span></span>
<span><span class="co">## [101] registry_0.5-1                    Biostrings_2.77.1                </span></span>
<span><span class="co">## [103] knitr_1.50                        gridExtra_2.3                    </span></span>
<span><span class="co">## [105] IRanges_2.43.0                    SummarizedExperiment_1.39.0      </span></span>
<span><span class="co">## [107] cmdfun_1.0.2                      stats4_4.5.0                     </span></span>
<span><span class="co">## [109] xfun_0.52                         Biobase_2.69.0                   </span></span>
<span><span class="co">## [111] brio_1.1.5                        matrixStats_1.5.0                </span></span>
<span><span class="co">## [113] DT_0.33                           stringi_1.8.7                    </span></span>
<span><span class="co">## [115] UCSC.utils_1.5.0                  lazyeval_0.2.2                   </span></span>
<span><span class="co">## [117] yaml_2.3.10                       evaluate_1.0.3                   </span></span>
<span><span class="co">## [119] codetools_0.2-20                  tibble_3.2.1                     </span></span>
<span><span class="co">## [121] cli_3.6.5                         systemfonts_1.2.3                </span></span>
<span><span class="co">## [123] processx_3.8.6                    jquerylib_0.1.4                  </span></span>
<span><span class="co">## [125] Rcpp_1.0.14                       GenomeInfoDb_1.45.3              </span></span>
<span><span class="co">## [127] emoji_16.0.0                      dbplyr_2.5.0                     </span></span>
<span><span class="co">## [129] XML_3.99-0.18                     parallel_4.5.0                   </span></span>
<span><span class="co">## [131] pkgdown_2.1.3                     ggplot2_3.5.2                    </span></span>
<span><span class="co">## [133] readr_2.1.5                       assertthat_0.2.1                 </span></span>
<span><span class="co">## [135] blob_1.2.4                        bitops_1.0-9                     </span></span>
<span><span class="co">## [137] viridisLite_0.4.2                 scales_1.4.0                     </span></span>
<span><span class="co">## [139] purrr_1.0.4                       crayon_1.5.3                     </span></span>
<span><span class="co">## [141] rlang_1.1.6                       TSP_1.2-5                        </span></span>
<span><span class="co">## [143] waldo_0.6.1</span></span></code></pre>
</details><p><br></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hiranyamaya Dash, Thomas Roberts, Maria Weinert, Nathan Skene.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
